test/TestProjects/MgmtDiscriminator/src/Generated/DeliveryRuleData.Serialization.cs

```diff
@@ -461,6 +477,33 @@ private BinaryData SerializeBicep(ModelReaderWriterOptions options)
                 }
             }
 
+            hasPropertyOverride = hasObjectOverride && propertyOverrides.TryGetValue("UnflattenedName", out propertyOverride);
+            if (Optional.IsDefined(Unflattened) || hasPropertyOverride)
+            {
+                builder.Append("  unflattened: ");
```

Comment:
If you do the above comment on line 481 this would need to go inside the if on 484 and the else on 501.